polaris/hub/storage.py

```diff
@@ -125,6 +129,149 @@ def _multipart_upload(self, key: str, value: bytes) -> None:
                 Bucket=self.bucket_name, Key=full_key, UploadId=upload_id, MultipartUpload={"Parts": parts}
             )
 
+    ## Custom methods
+
+    def copy_to_destination(
+        self, destination: Store, if_exists: ZarrConflictResolution = "replace", log: Callable = lambda: None
+    ) -> tuple[int, int, int]:
+        """
+        Copy the content of this store to the destination store.
+
+        We leverage the internal knowledge of this store to make the operation more efficient than `zarr.copy_store`:
+            - Parallel, concurrent `getitems` operations using a thread pool
```

Comment:
**Question:** I'm not seeing the parallelism here...? What am I missing? This code feels very sequential to me.

Reply:
`getitems` is used to fetch multiple items concurrently and in parallel.